### PR TITLE
add related resource links

### PR DESCRIPTION
**Pull request by sean1588**
registry pr: https://github.com/pulumi/registry/pull/1733

figma: https://www.figma.com/file/bVEoSwAvU7KDpuPNri332F/registry-v2-compact?node-id=127%3A2778&t=J4HRRRRuJChcQaFn-0

This PR adds the stencil component to render the links related to a particular resource.

preview from registry:
http://registry-origin-pr-1733-9c20f36b.s3-website.us-west-2.amazonaws.com/registry.

I limited the links to a maximum of 7, though am open to any other suggestions. Also, for right now we are only producing links for aws, gcp, and azure (both native and classic versions). We can support more packages in the future. I left them out for now since there are some oddities with other packages and the url routes that they live at, so the url is not always deterministic based on the package and module names.

image

I also made a change to move the vertical scroll bars under the package card. Currently when the TOC expands beyond the height of the screen we end up with a vertical scroll bar on the right nav on some pages. This is now more common, now that we have the related links in the nav also. I adjusted it so the scrolling area is below the card when this happens. See screenshots:

current: 
Screenshot from 2022-12-20 10-29-45

with change:
Screenshot from 2022-12-20 10-30-02

### CODE REVIEW

**Comment by susanev** [comment]
the request a change link looks really out of place where it is with the addition of this section; @anita-trimbur would be great to get your thoughts on the design of this right panel

**Comment by sean1588** [comment]
yeah I'd love to get your thoughts on the right nav.  yeah I think I may need to move that request a change link below the related resources section.

**Comment by anita-trimbur** [comment]
for sure i can take a look at the design side today!

**Comment by sean1588** [comment]
oh wow that was fast! thanks @anita-trimbur! looks great!!

**Comment by sean1588** [comment]
@anita-trimbur would you be ok with me just moving the "request a change" link below the related resources section for this PR and will leave the other pieces as is matching what we currently have. I can then do a fast follow to address the styling change to add that nice left bar you have for the indentation and update the fonts in a follow up PR. My main reasoning for this is the other section above the resource links is used also on other registry pages, not just the individual resource page, so I would rather address the new styling separately to keep this PR focused on the changes to this specific page.

@zchase , I updated this to pull the json from a github gist I created for now. I need to get the JSON file added to that s3 bucket we have and then update the url to fetch from there. Can you help me get that file uploaded to that bucket?

**Comment by anita-trimbur** [comment]
for sure that's fine by me!

**Comment by anita-trimbur** [comment]
@sean1588 not sure how quick a change this is, but @susanev made a good point that we should align with the new h1 format, so `ec2.SecurityGroup` instead of `ec2 / SecurityGroup`

File: stencil/src/components/resource-links/resource-links.scss [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+        font-size: 14px;
+        padding-bottom: 2px;
+    }
+    .links {
+        padding-left: 10px;
+        padding-top: 8px;
+        font-size: 14px;
+        color: #2e3872;
+        a:hover {
+            color: #414ebc;
```
Comment (sean1588) [inline stencil/src/components/resource-links/resource-links.scss:17]: I cannot seem to locate where these colors exist. I got them by inspecting the elements in the TOC. I do not see anywhere where they are defined though, which is why I hard coded them.

File: stencil/src/components/resource-links/resource-links.scss [hunk 2]
```diff
@@ -0,0 +1,20 @@
+pulumi-resource-links{
+    .container {
+        margin-bottom: 24px;
+    }
+    .heading {
+        color: #2e3872;
+        font-weight: 600;
```
Comment (sean1588) [inline stencil/src/components/resource-links/resource-links.scss:7]: Are we able to use tailwind styles in stencil components here? When I use @apply they dont seem to take effect. I can inline the tailwind classes in the mark up if that is the case.

File: stencil/src/components/resource-links/resource-links.tsx [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+        // Fetch JSON file containing resource link information.
+        fetch("https://www.pulumi.com/uploads/related-resources/2022-12-15.json")
+        .then(resp => resp.json())
+        .then((response) => {
+            this.relatedResources = response;
+        })
+        .catch(err => console.error(err))
+    }
+
+    render() {
```
Comment (zchase) [inline stencil/src/components/resource-links/resource-links.tsx:33]: The logic here could arguably also be done in `componentWillLoad`. That way this logic doesn't run on every render. Also you wouldn't have to store the entire resource response in the state, you could just store the related resources (if there is one).

**Review by zchase — APPROVED** [review]
LGTM --- two comments but both aren't blocking.

**Review by anita-trimbur — APPROVED** [review]
🚀

**Pull request merged**